syntax = "proto2";
import "comm_message.proto";

//登录请求
message LoginRequest
{
	required int32 channel = 1; //渠道，1：winTest
	required string openid = 2; //open_id,由服务器生成的账号唯一id
	required string key = 3; //通过cgi获取的key
}

//登录应答
message LoginAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	required uint32 openid = 2; //账号唯一id
}

//获取角色列表应答
message PlayerListAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	repeated PlayerBaseInfo playerList = 2; //角色列表
}

//创建角色请求
message PlayerCreateRequest
{
	required string name = 1; //角色名
	required uint32 job = 2; //角色职业
	required uint32 sex = 3; //性别，0男，1女
	required uint32 default_hair = 4;
	required uint32 default_icon = 5;
	required uint32 default_hair_color = 6;
	required uint32 default_up_color = 7;
	required uint32 default_down_color = 8;
}

//创建角色应答
message PlayerCreateAnswer
{
	required uint32 result = 1; //结果，0：成功，否则失败
	optional PlayerBaseInfo playerList = 2; //角色信息
}

//进入游戏请求
message EnterGameRequest
{
	required uint64 playerId = 1; //选择的角色id
	required uint32 reconnect = 2; //是否断线重连，0：正常登陆，1：断线重连
}

//进入游戏应答
message EnterGameAnswer
{
	required uint32 result = 1; //结果
	required float posX = 2; //X坐标
	required float posZ = 3; //Z坐标
	required uint32 sceneId = 4; //场景id
	required uint32 curTime = 5; //服务器当前时间（秒）
	required int32 posY = 6; //Y坐标  有坐骑的时候需要用
	required uint32 CurHorse = 7; //当前坐骑id
	required uint32 CurHorseState = 8; //当前坐骑乘骑状态 0不在坐骑上  1在坐骑上
	required int32 fly = 9; //2飞行 1陆行
	required int32 direct = 10;	  //朝向
	required uint32 horse_step = 11; //当前坐骑阶
}

message HeartbeatNotify
{
	required uint32 curTime = 1; //服务器当前时间（秒）	
}

//账号服务器信息
message AccountServerData
{
	required uint32 serverId = 1; //服务器ID
	repeated PlayerBaseInfo playerList = 2; //角色列表
	required uint32 lastLoginTime = 3; //最后登录时间
}

//账号服务器列表
message AccountServerListAnswer
{
	repeated AccountServerData serverList = 2; //服务器列表
	required uint64 cur_server_time = 3;  //当前服务器时间
}

